subsys/net/lib/sockets/sockets.c

```diff
@@ -517,24 +614,20 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
 		}
 	}
 
-	/* Set starting point behind packet header since we've
-	 * handled src addr and port.
-	 */
-	header_len = net_pkt_appdata(pkt) - pkt->frags->data;
-
-	recv_len = net_pkt_appdatalen(pkt);
+	recv_len = net_pkt_data_to_read(pkt);
 	if (recv_len > max_len) {
 		recv_len = max_len;
 	}
 
-	/* Length passed as arguments are all based on packet data size
-	 * and output buffer size, so return value is invariantly == recv_len,
-	 * and we just ignore it.
-	 */
-	(void)net_frag_linearize(buf, recv_len, pkt, header_len, recv_len);
+	if (net_pkt_read_new(pkt, buf, recv_len)) {
+		errno = -ENOBUFS;
```

Comment:
Remove that "-" from -ENOBUFS. errno is not -ve.

Reply:
good catch